docs(boards): add docs (#4725)
gnolang/gno: examples/gno.land/r/gnoland/boards2/v1/README.md

> # Boards: Decentralized Discussion Forums on Gno.land

Reviewer: Maybe?

```suggestion
# Boards: Decentralized Discussions on Gno.land
```

Boards is a forum realm for Gno.land with robust role-based access control, moderation tools, and a clean public API. It serves two roles: a proof-of-concept that demonstrates building real dApps on Gno.land, and a practical tool for decentralized social coordination on the internet. It features pluggable permission systems, on-chain governance with irreversible commitments, moderation that balances openness with safety, and an event-driven architecture that enables integrations and observability. For communities, it provides transparent governance structures, on-chain decision-making, configurable moderation, permanent auditable records, and flexible role management that bridges on-chain governance with human dynamics.

> Access (v1): Boards operates invite-only access during beta. Creating boards and other privileged actions require realm membership. How to request/receive an invite: [TO ADD].

Reviewer: Right now Boards2 exposes a [RequestInvite()](https://gno.land/r/gnoland/boards2/v1:admin-users$help#func-RequestInvite) function and a view that display invites requests that need approval thought the "Manage Board -> List Invite Requests" menu in each individual board.

I'm going add a top "Request Invite" link to each board.

- Boards organizes content into boards → threads → replies
- Each board has its own permissions (roles, thresholds, ban-list, freeze state)
- Realm-level governance can be locked permanently and independently from member modifications
- Moderation leverages per-board flagging thresholds, freezing, and bans
- Permissions are enforced via a pluggable `Permissions` interface; the default implementation is `BasicPermissions` backed by a DAO member set

### Key Packages and Files

- Public API: `examples/gno.land/r/gnoland/boards2/v1/public.gno`
- Board model and helpers: `examples/gno.land/r/gnoland/boards2/v1/board.gno`
- Post model and helpers: `examples/gno.land/r/gnoland/boards2/v1/post.gno`
- Permissions: `examples/gno.land/r/gnoland/boards2/v1/permissions.gno`, `permissions_basic.gno`, `permissions_validators.gno`
- Moderation (flag/ban/freeze): `public_flag.gno`, `flag.gno`, `public_ban.gno`, `public_freeze.gno`
- Realm lock: `public_lock.gno`
- Invitations: `public_invite.gno`
- Rendering helpers/URIs: `render.gno`, `uris_post.gno`

## Roles and Permissions

> Authorization model controls who can do what, and where. Boards uses a pluggable permission engine to enforce capabilities at realm and board scopes. Use the default mapping, refine with validators, or swap the engine with `SetPermissions` to fit governance.

Reviewer: Maybe?

```suggestion
Authorization model controls who can do what, and where. Boards uses a pluggable permission engine to enforce capabilities at realm and board scopes. When the default engine is not adequate it can be swapped to a custom one with `SetPermissions(...)` to fit governance.
```
